src/HttpClientFlexible.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace Http\Client\Common;
+
+use Http\Client\HttpAsyncClient;
+use Http\Client\HttpClient;
+
+/**
+ * A flexible http client, which implements both interface and will emulate
+ * one contract, the other, or none at all depending on the injected client contract.
+ *
+ * @author Joel Wurtz <joel.wurtz@gmail.com>
+ */
+class HttpClientFlexible implements HttpClient, HttpAsyncClient
```

Comment:
i think this name is a bit weird. maybe HttpSyncAsyncClient ? or any better ideas?